Objective-C/CBLDatabase.mm

```diff
@@ -790,13 +814,107 @@ - (void) freeC4DB {
 }
 
 
-#pragma mark - DOCUMENT SAVE AND CONFLICT RESOLUTION
+#pragma mark - DOCUMENT SAVE AND CONFLICT HANDLING
+
+
+- (BOOL) saveDocument: (CBLDocument*)document
+   concurrencyControl: (CBLConcurrencyControl)concurrencyControl
+           asDeletion: (BOOL)deletion
+                error: (NSError**)outError
+{
+    if (deletion && !document.revID) {
+        [NSException raise: NSInvalidArgumentException
+                    format: @"Do not allow to delete a newly created document "
+         "that has not been saved into the database."];
+        return NO;
+    }
+    
+    CBL_LOCK(self) {
+        if (![self prepareDocument: document error: outError])
+            return NO;
+        
+        // Begin a db transaction:
+        C4Transaction transaction(_c4db);
+        if (!transaction.begin())
+            return convertError(transaction.error(), outError);
+        
+        C4Document* curDoc = nil;
+        if (deletion) {
+            // Check existing, NO-OPS if the document doesn't exist:
```

Comment:
I don't think you need to make this extra check; won't the `-saveDocument:into:...` call fail with a conflict, and you can detect that?

Reply:
Currently when saving a deleted doc to non-existing doc, LC returns error code 15 (invalid parameter). I'm not sure that is the right approach to check the error code 15 and no-ops.

Reply:
Would it make sense to have LC return NOT_FOUND error instead so we can check the error code for this case? The current INVALID_ARGUMENT (15) error code returned is not clear.

Reply:
Hmm, I just wrote a unit test of calling `c4_put` to save a deletion revision to a nonexistent doc, and it returns kC4ErrorNotFound.

Reply:
And `c4_update` of a purged document returns `kC4ErrorConflict`. What are the details of the call you make that returns `kc4InvalidArgument`?